odbc_fdw.c

```diff
@@ -174,7 +177,62 @@ static struct odbcFdwOption valid_options[] =
 	{ NULL,       InvalidOid}
 };
 
-typedef enum { TEXT_CONVERSION, HEX_CONVERSION, BIN_CONVERSION, BOOL_CONVERSION } ColumnConversion;
+typedef enum { TEXT_CONVERSION, BIN_CONVERSION, BOOL_CONVERSION } ColumnConversion;
+
+static GetDataTruncation
+result_truncation(SQLRETURN ret, SQLHSTMT stmt)
+{
+    SQLCHAR sqlstate[5];
```

Comment:
Shouldn't this be 6 (5 + NULL)?

```
SQLState
[Output] Pointer to a buffer in which to return a five-character SQLSTATE code (and terminating NULL) for the diagnostic record RecNumber. The first two characters indicate the class; the next three indicate the subclass. This information is contained in the SQL_DIAG_SQLSTATE diagnostic field. For more information, see SQLSTATEs.
```

Reply:
Ouch, good catch, it must have been working by chance:  the next allocated element in the stack frame contains a zero value (`NO_TRUNCATION` which must have been overwritten with a zero byte) before the string comparisons were made. 🙄